include/inexor/vulkan-renderer/wrapper/descriptor_builder.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <cassert>
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+class Device;
+class ResourceDescriptor;
+
+class DescriptorBuilder {
+    const Device &m_device;
+    const std::uint32_t m_swapchain_image_count{};
+
+    std::vector<VkDescriptorSetLayoutBinding> m_layout_bindings{};
+    std::vector<VkWriteDescriptorSet> m_write_sets{};
+
+    std::vector<VkDescriptorBufferInfo> m_descriptor_buffer_infos{};
+    std::vector<VkDescriptorImageInfo> m_descriptor_image_infos{};
+
+public:
+    /// @brief Constructs the descriptor builder.
+    /// @param device The const reference to a device RAII wrapper instance.
+    /// @param swapchain_image_count The number of images in swapchain.
+    DescriptorBuilder(const Device &device, const std::uint32_t swapchain_image_count);
+
+    DescriptorBuilder(const DescriptorBuilder &) = delete;
+    DescriptorBuilder(DescriptorBuilder &&) noexcept;
```

Comment:
```suggestion
    DescriptorBuilder(DescriptorBuilder &&) = delete;
```
This doesnt't look like it's implemented anyway?

Reply:
Right! Done.